File: dcp-solutions/2024-10-26/solution.rb
```ruby
def max_profit(prices)
  return 0 if prices.empty?
  min_price = prices[0]
  max_profit = 0
  
  prices.each do |price|
    min_price = price if price < min_price
    profit = price - min_price
    max_profit = profit if profit > max_profit
  end
  
  max_profit
end
```

File: dcp-solutions/2024-10-26/solution.php
```php
function maxProfit($prices) {
    if (empty($prices)) {
        return 0;
    }

    $minPrice = $prices[0];
    $maxProfit = 0;

    foreach ($prices as $price) {
        $minPrice = min($minPrice, $price);
        $maxProfit = max($maxProfit, $price - $minPrice);
    }

    return $maxProfit;
}
```